scripts/resource/class_generator.py

```diff
@@ -0,0 +1,244 @@
+        type_from_dict_for_init = 'Optional[str] = ""'
+
+    if type_from_dict == "bool":
+        type_from_dict_for_init = "Optional[bool] = None"
+
+    if type_from_dict == "int":
+        type_from_dict_for_init = "Optional[int] = None"
+
+    return (
+        name,
+        f"{name}: {type_from_dict_for_init}",
+        required,
+        type_from_dict,
+    )
+
+
+def generate_resource_file_from_dict(resource_dict: Dict[str, Any], output_dir="ocp_resources") -> str:
+    env = Environment(
+        loader=FileSystemLoader("scripts/resource/manifests"),
+        trim_blocks=True,
+        lstrip_blocks=True,
+        undefined=DebugUndefined,
+        autoescape=True,
+    )
+
+    template = env.get_template(name="class_generator_template.j2")
+    rendered = template.render(resource_dict)
+    undefined_variables = meta.find_undeclared_variables(env.parse(rendered))
+    if undefined_variables:
+        LOGGER.error(f"The following variables are undefined: {undefined_variables}")
+        raise click.Abort()
+
+    temp_output_file: str = ""
+    output_file = f"{output_dir}/{format_resource_kind(resource_kind=resource_dict['KIND'])}.py"
+    if os.path.exists(output_file):
+        temp_output_file = f"{output_file[:-3]}_TEMP.py"
+        LOGGER.warning(f"{output_file} already exists, using {temp_output_file}")
+        output_file = temp_output_file
+
+    with open(output_file, "w") as fd:
+        fd.write(rendered)
+
+    for op in ("format", "check"):
+        run_command(
+            command=shlex.split(f"poetry run ruff {op} {output_file}"),
+            verify_stderr=False,
+        )
+
+    return output_file
+
+
+def resource_from_explain_file(file: str, namespaced: bool, api_link: str) -> Dict[str, Any]:
+    section_data: str = ""
+    sections: List[str] = []
+    resource_dict: Dict[str, Any] = {
+        "BASE_CLASS": "NamespacedResource" if namespaced else "Resource",
+        "API_LINK": api_link,
+    }
+    new_sections_words: Tuple[str, str, str] = ("KIND:", "VERSION:", "GROUP:")
+
+    with open(file) as fd:
+        data = fd.read()
+
+    for line in data.splitlines():
+        # If line is empty section is done
+        if not line.strip():
+            if section_data:
+                sections.append(section_data)
+                section_data = ""
+
+            continue
+
+        section_data += f"{line}\n"
+        if line.startswith(new_sections_words):
+            if section_data:
+                sections.append(section_data)
+                section_data = ""
+            continue
+
+    # Last section data from last iteration
+    if section_data:
+        sections.append(section_data)
+
+    start_fields_section: str = ""
+
+    for section in sections:
+        if section.startswith("FIELDS:"):
+            start_fields_section = section
+            continue
+
+        key, val = section.split(":")
+        resource_dict[key.strip()] = val.strip()
+
+    resource_dict["SPEC"] = []
```

Comment:
This is a bit confusing, as actual object spec is a dict. Why not use a dict instead?

Reply:
This is not a representation of the resource spec, this is the parser of explain output for the jinja template.